app/src/main/java/com/goldenraven/padawanwallet/ui/drawer/SettingsScreen.kt

```diff
@@ -1,15 +1,116 @@
 package com.goldenraven.padawanwallet.ui.wallet
 
+import android.content.Context
+import android.util.Log
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.SnackbarHost
+import androidx.compose.material.SnackbarHostState
 import androidx.compose.material3.*
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
 import androidx.navigation.NavController
+import com.goldenraven.padawanwallet.BuildConfig
+import com.goldenraven.padawanwallet.R
+import com.goldenraven.padawanwallet.data.Repository
+import com.goldenraven.padawanwallet.data.Wallet
 import com.goldenraven.padawanwallet.ui.DrawerAppBar
-
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.features.auth.*
+import io.ktor.client.features.auth.providers.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.ktor.http.content.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 internal fun SettingsScreen(navController: NavController) {
+
+    val faucetCallDone = Repository.wasOneTimeFaucetCallDone()
+    val coroutineScope = rememberCoroutineScope()
+    val snackbarHostState = remember { SnackbarHostState() }
+    val context = LocalContext.current
+
     Scaffold(
         topBar = { DrawerAppBar(navController, title = "Settings") },
-    ) { }
+        bottomBar = { SnackbarHost(hostState = snackbarHostState) }
+    ) {
+        Column(
+            Modifier.padding(all = 16.dp)
+        ) {
+            if (faucetCallDone) {
+                Button(
+                    onClick = {
+                        callTatooineFaucet(Wallet.getNewAddress(), coroutineScope, snackbarHostState, context)
+                    },
+                    modifier = Modifier.fillMaxWidth()
+                ) {
+                    Text(text = stringResource(id = R.string.request_testnet_coins))
+                }
+            } else {
+                Button(
+                    onClick = {
+                        coroutineScope.launch {
+                            snackbarHostState.showSnackbar(context.resources.getString(R.string.errorFaucet))
```

Comment:
In the case where the user has already done their call to the faucet to receive some testnet coins, the button should be grayed out (as you currently have it) and the snackbar should instead read something like "Error: you've already received testnet coins!".

Or maybe just nothing happens and the button is grayed out. What do you think? Open to both versions.

Reply:
I think it would be good to let the user know why the button is greyed out, which is showing the snackbar. Though I'm open to both versions too.

Reply:
Sounds good let's show the snackbar indicating they have already used their one-time faucet pull.